### DISCUSSION TITLE
Import/export option

### DISCUSSION THREAD

**Discussion by David42fr** [teslamate-org/teslamate#1998] [Ideas]
Hi, 

New to Teslamate and Github here ;)

I have 3 instances of Teslamate running and some drives or charges are well registered in one instance and not in another one.

It should be great to be able to export the drive/charge from an instance and import it on another one?

Maybe it's really difficult implementing it?

thanks for reading

**Comment by DrMichael** [comment]
You might find https://docs.teslamate.org/docs/maintenance/backup_restore helpful...

**Reply by David42fr** [reply]
Thanks, but I have 2 problems :

1. You cannot choose which datas you want to keep or bin : my first instance was a testing purpose one and have the first drive I record. I would like to import this particular one to the second instance on my PC or on the instance on Oracle. But there's maybe a set of SQL commands I can type?
2. I have 2 instances in docker-widows and one in Oracle. When I try to backup/restore, I have a problem with the file encoding type. I have to put the backup file in UTF8 mode but then have troubles with the restore process.

Nice tool anyway ;)

**Reply by David42fr** [reply]
Here is the error code I got : 
`
ERROR:  invalid byte sequence for encoding "UTF8": 0xff
`
And if I open the file with notepad++ and convert it in UTF8, the size is half the original one and all data have problems with é è and those sort of characters.

**Comment by itachi46** [comment]
I have this requirement as well, my initial teslamate instance died one day (hardware issues) so I reinstalled from scratch. later on I was able to recover the initial database I'm not sure how to merge the two databases now?

the instructions at https://docs.teslamate.org/docs/maintenance/backup_restore is for full DB restore (i.e. drop all current data and replace with old) is there a way to merge both databses?

**Reply by eliluong** [reply]
There is no automated way currently. You can, however, merge the databases manually, but you'd have to study the database structure to make sure to increment certain values correctly, make sure all the geofence_id and locations match up between the two databases, among other things.
